presto-benchmark-runner/src/main/java/com/facebook/presto/benchmark/framework/BenchmarkManager.java

```diff
@@ -0,0 +1,173 @@
+import java.util.concurrent.ExecutorCompletionService;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.facebook.presto.benchmark.event.BenchmarkQueryEvent.EventStatus;
+import static com.facebook.presto.benchmark.event.BenchmarkQueryEvent.EventStatus.FAILED;
+import static com.facebook.presto.benchmark.event.BenchmarkQueryEvent.EventStatus.SUCCEEDED;
+import static com.facebook.presto.benchmark.framework.PhaseSpecification.ExecutionStrategy.CONCURRENT;
+import static com.facebook.presto.benchmark.runneraction.BenchmarkQueryExceptionClassifier.shouldResubmit;
+import static java.lang.Thread.currentThread;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.Executors.newFixedThreadPool;
+
+public class BenchmarkManager
+        implements BenchmarkQueryResubmitter
+{
+    private static final Logger log = Logger.get(BenchmarkManager.class);
+
+    private final BenchmarkSuiteSupplier benchmarkSuiteSupplier;
+    private final BenchmarkQueryRunnerFactory benchmarkQueryRunnerFactory;
+    private final int maxConcurrency;
+    private ExecutorService executor;
+    private CompletionService<Optional<BenchmarkQueryEvent>> completionService;
+    private final Set<EventClient> eventClients;
+    private AtomicInteger queriesSubmitted = new AtomicInteger();
+
+    private int benchmarkResubmissionLimit;
+    private Map<String, Integer> resubmittedCounts = new ConcurrentHashMap<>();
+
+    @Inject
+    public BenchmarkManager(
+            BenchmarkSuiteSupplier benchmarkSuiteSupplier,
+            BenchmarkQueryRunnerFactory benchmarkQueryRunnerFactory,
+            BenchmarkRunnerConfig config,
+            Set<EventClient> eventClients)
+    {
+        this.benchmarkSuiteSupplier = requireNonNull(benchmarkSuiteSupplier, "benchmarkSuiteSupplier is null");
+        this.benchmarkQueryRunnerFactory = requireNonNull(benchmarkQueryRunnerFactory, "benchmarkQueryRunnerFactory is null");
+        this.maxConcurrency = config.getMaxConcurrency();
+        this.benchmarkResubmissionLimit = config.getBenchmarkResubmissionLimit();
+        this.eventClients = ImmutableSet.copyOf(eventClients);
+    }
+
+    @PostConstruct
+    public void start()
+    {
+        this.executor = newFixedThreadPool(maxConcurrency);
+        this.completionService = new ExecutorCompletionService<>(executor);
+        BenchmarkSuite benchmarkSuite = benchmarkSuiteSupplier.get();
+        Map<String, BenchmarkQuery> queries = benchmarkSuite.getQueryMap();
+        BenchmarkSuiteInfo benchmarkSuiteInfo = benchmarkSuite.getSuiteInfo();
+        List<PhaseSpecification> phases = benchmarkSuiteInfo.getPhases();
+        for (PhaseSpecification phase : phases) {
+            if (phase.getExecutionStrategy() == CONCURRENT) {
+                submit(ImmutableList.copyOf(queries.values()), benchmarkSuiteInfo.getSessionProperties());
+            }
+        }
```

Comment:
This is an incorrect logic.
1. Each phase needs to be run one after another. `submit` is a non-blocking call. Do a for loop here will submit all queries in all phases.
2. Since this PR is not support stream execution yet. We should throw if there is any phases with stream execution.
3. We need construct and use the proper executor here. `BenchmarkManager` should need to know how to submit/execute those queries.